sdcm/scan_operation_thread.py

```diff
@@ -437,6 +437,9 @@ def __init__(self, generator, **kwargs):
         super().__init__(generator, scan_event=FullScanAggregateEvent, **kwargs)
         self._session_execution_timeout = self.fullscan_params.full_scan_aggregates_operation_limit + 10*60
 
+        self.validate_forward_service_requests_counter = 0
+        self.validate_forward_service_requests_start_time = time.time()
```

Comment:
this should be reset in `execute_query` because currently it will be verifying this metric since `ScanOperationThread` start, basically, always since beginning of the test

Reply:
Could you please explain in more detailed why you think so?

according to lo the logs it works as expected: at the resetting counter I reset the next request "start_time" to now and all request has no intersections
```
 {'metric': {'__name__': 'scylla_forward_service_requests_dispatched_to_other_nodes', 'instance': '10.4.0.170', 'job': 'scylla', 'shard': '3'}, 'values': [[1690371602.091, '0'], [1690371622.091, '0'], [1690371642.091, '0'], [1690371662.091, '0'], [1690371682.091, '0'], [1690371702.091, '0'], [1690371722.091, '0'], [1690371742.091, '0'], [1690371762.091, '5']]}
```
```
{'metric': {'__name__': 'scylla_forward_service_requests_dispatched_to_other_nodes', 'instance': '10.4.0.143', 'job': 'scylla', 'shard': '2'}, 'values': [[1690371957.984, '20'], [1690371977.984, '25'], [1690371997.984, '25'], [1690372017.984, '25'], [1690372037.984, '25'], [1690372057.984, '25']]}
```
```
{'metric': {'__name__': 'scylla_forward_service_requests_dispatched_to_other_nodes', 'instance': '10.4.0.143', 'job': 'scylla', 'shard': '2'}, 'values': [[1690372066.284, '25'], [1690372086.284, '25'], [1690372106.284, '30'], [1690372126.284, '30'], [1690372146.284, '30'], [1690372166.284, '35']]}
```

Reply:
it's not the only place this is being assigned to
after every query it's being assigned again

Reply:
I think this should be assigned after every query to prometheus , but before every CQL command, same as log marks are used, the range like that can be too big, since the last time a scan was done.

Reply:
I overlooked it.
But acutally I expected it to be reset before executing a query - so placing it next to query seems more logical for me.
IIUC, we reset this timer, run query 5 times and then verify if this counter went up. Is it what was intended?

Reply:
@soyacz Yes 1 scan operation may not increment metric ( at the beginning of the test for sure) 

@fruch 
that is why I collect metrics in the time range between 5 scan operations and check that Metrick changed in this time range. yes, a range like that can be too big. But for now, I can find a way how to check it more properly.

Previously it was checked per 1 command but with a WARNING, because the scan operation may not have "Dispatching forward_request' message in the trace also

Reply:
> @fruch that is why I collect metrics in the time range between 5 scan operations and check that Metrick changed in this time range. yes, a range like that can be too big. But for now, I can find a way how to check it more properly.

you can do it (2 ideas):
1. run single query (even not a time range) and save values (or even just one value - the sum) in memory, after scan operation, query the same again and compare numbers
2. don't even use Prometheus - could be done by querying metrics for each node separately one by one using GET HTTP requestand saving the metrics (also sum should do well), but this could be less stable.

Possibly this approach could solve issue with missing metric - in case first query didn't return it, assume 0, the next one after first scan operation should appear.

Reply:
@soyacz 

again the problem is:  1 scan operation may not increment metric
it is updated periodically(not immediately) with minimal increment 5 (according to what I see)

it does not matter if you request a time range or not. In time rage I just took 1st and last value

here is the metrics graph:
![Screenshot from 2023-07-27 12-44-14](https://github.com/scylladb/scylla-cluster-tests/assets/1948660/7c918037-961f-46f7-ac1f-027f2f7558f5)

Reply:
> Possibly this approach could solve issue with missing metric - in case first query didn't return it, assume 0, the next one after first scan operation should appear.

it always returns metric. the problem is that it is may  not increment it after 1 scan operation

Reply:
this graph comes from 6 nodes test? this would explain increment by 5
Also, I see 22 increments - how many aggregate queries were run?

Reply:
tih is metrics from some random job.


bf974e95-483a-45ff-8e2f-56000d042c9d - this is my short term test job with nemesis_class_name: 'NoOpMonkey' and run_fullscan: ['{"mode": "aggregate"...

it has 41 FullScanAggregate executions 

![Screenshot from 2023-07-27 12-58-02](https://github.com/scylladb/scylla-cluster-tests/assets/1948660/5be1df0f-0994-42b6-b3ce-9c84b4ebada4)

Reply:
@soyacz 
look like you right I need to just wait about 30s. set retry on 60s and looks like it works, I need more time for testing but code itself is ready for review

cc @fruch